### ISSUE TITLE
Improve GenerateModel()

### ISSUE THREAD

**Issue by steinbohrer** [AdrianStrugala/AvroConvert#58]
If you take the following schema to generate a C# model:

```
{
  "type": "record",
  "name": "Result",
  "fields": [
    {
      "name": "module",
      "type": {
        "type": "record",
        "name": "module",
        "fields": [
          {
            "name": "moduleIdName",
            "type": [
              "null",
              "string"
            ],
            "default": null
          },
          {
            "name": "data",
            "type": {
              "type": "array",
              "items": {
                "type": "record",
                "name": "module",
                "namespace": "data",
                "fields": [
                  {
                    "name": "isControl",
                    "type": [
                      "null",
                      "boolean"
                    ],
                    "default": null
                  }
                ]
              }
            }
          },
          {
            "name": "instrumentId",
            "type": [
              "null",
              "int"
            ],
            "default": null
          }
        ]
      }
    },
    {
      "name": "customerInfo",
      "type": {
        "type": "record",
        "name": "customerInfo",
        "fields": [
          {
            "name": "country",
            "type": "string",
            "doc": ""
          },
          {
            "name": "customerNumber",
            "type": "string",
            "doc": ""
          }
        ]
      },
      "doc": ""
    },
    {
      "name": "deviceId",
      "type": "string"
    },
    {
      "name": "url",
      "type": "string",
      "doc": ""
    }
  ]
}
```

you will get two classes named "module" which makes the code not directly usable without manually renaming.

A simple solution to this would be to add the namespace, if there is one.

So, I changed:

```
                        AvroClass c = new AvroClass()
                        {
                            ClassName = parent["name"].ToString().Split('.').Last()
                        };
                        if (parent["namespace"] != null)
                        {
                            c.ClassName = parent["namespace"].ToString() + c.ClassName;
                        }
                        classes.Add(c);
```

and

```
private static string ResolveArray(JObject typeObj)
        {
            string fieldType;
            string fieldNamespace = null;

            // If this is an array of a specific class that's being defined in this area of the json
            if (typeObj["items"] is JObject && ((JObject)typeObj["items"])["type"].ToString() == "record")
            {
                fieldType = ((JObject)typeObj["items"])["name"].ToString();
                fieldNamespace = ((JObject) typeObj["items"])["namespace"]?.ToString();
            }
            else if (typeObj["items"] is JValue value)
            {
                fieldType = value.ToString();
            }
            else
            {
                throw new InvalidAvroObjectException($"{typeObj}");
            }

            if (!string.IsNullOrWhiteSpace(fieldNamespace))
            {
                fieldType = fieldNamespace + fieldType;
            }

            fieldType += "[]";

            return fieldType;
        }
```

This will give me a "module" and a "datamodule" class, so I can directly copy that code into my program.

Bu there might be other possible places where a namespace could appear.

**Comment by AdrianStrugala** [comment]
Good morning,

Thank you for reporting the problem. I need to take a look at how namespaces could be handled in model generation. Your proposition makes sense, but I need to test also other possibilities.

Regards,
Adrian

**Comment by AdrianStrugala** [comment]
Hello,

The issue is resolved. The fix is ready in 3.2.1 version of AvroConvert

Thanks,
Adrian
